src/rust/engine/src/nodes.rs

```diff
@@ -1069,6 +1070,13 @@ impl Node for NodeKey {
       | NodeResult::Value(_) => None,
     }
   }
+
+  fn cacheable(&self) -> bool {
```

Comment:
As discussed in the ticket: it would be totally fine to mark this method (and anything else that needs it) as `#[allow(dead_code)]` and defer actually using the `cacheable` flag for another PR. Up to you!